src/components/clusterWizard/ClusterDetails.tsx

```diff
@@ -0,0 +1,273 @@
+            cannot be changed after cluster installation. The full cluster address will be: <br />
+            <strong>
+              {clusterName || '[Cluster Name]'}.{baseDnsDomain || '[example.com]'}
+            </strong>
+          </>
+        );
+
+        const toggleRedHatDnsService = (checked: boolean) =>
+          setFieldValue('baseDnsDomain', checked ? managedDomains.map((d) => d.domain)[0] : '');
+
+        const form = (
+          <Grid hasGutter>
+            <GridItem span={12} lg={10} xl={9} xl2={7}>
+              <Form
+                id="wizard-cluster-details__form"
+                onKeyDown={(e) => {
+                  if (e.key === 'Enter') {
+                    submitForm();
+                  }
+                }}
+              >
+                <InputField ref={nameInputRef} label="Cluster Name" name="name" isRequired />
+                <SelectField
+                  label="OpenShift Version"
+                  name="openshiftVersion"
+                  options={[{ label: openshiftVersion, value: openshiftVersion }]}
+                  // getHelperText={getOpenshiftVersionHelperText}
+                  isDisabled
+                  isRequired
+                />
+                {(pullSecret || !pullSecretSet) && <PullSecret pullSecret={pullSecret} />}
+                {!!managedDomains.length && (
+                  <CheckboxField
+                    name="useRedHatDnsService"
+                    label="Use a temporary 60-day domain"
+                    helperText="A base domain will be provided for temporary, non-production clusters."
+                    onChange={toggleRedHatDnsService}
+                  />
+                )}
+                {values.useRedHatDnsService ? (
+                  <SelectField
+                    label="Base Domain"
+                    name="baseDnsDomain"
+                    helperText={baseDnsHelperText}
+                    options={managedDomains.map((d) => ({
+                      label: `${d.domain} (${d.provider})`,
+                      value: d.domain,
+                    }))}
+                    isRequired
+                  />
+                ) : (
+                  <InputField
+                    label="Base Domain"
+                    name="baseDnsDomain"
+                    helperText={baseDnsHelperText}
+                    placeholder="example.com"
+                    isDisabled={useRedHatDnsService}
+                    isRequired
+                  />
+                )}
+              </Form>
+            </GridItem>
+          </Grid>
+        );
+        const footer = (
+          <Stack hasGutter>
+            {!!alerts.length && (
+              <StackItem>
+                <Alerts />
+              </StackItem>
+            )}
+            <StackItem>
+              <ClusterToolbar>
+                <ToolbarButton
+                  name="save"
```

Comment:
This logic + component composition are repeating on every step, so they are co-located in the `ClusterWizardToolbar`.

Reply:
Yes, I'll conform to it in a follow up.